app/controllers/partners/PartnerTelephoneNumberController.scala

```diff
@@ -0,0 +1,85 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.partners
+
+import config.{BaseControllerComponents, FrontendAppConfig}
+import controllers.BaseController
+import forms.PartnerTelephoneForm
+import models.api.Individual
+import play.api.mvc.{Action, AnyContent}
+import services.EntityService.Telephone
+import services._
+import uk.gov.hmrc.auth.core.AuthConnector
+import views.html.partners.PartnerTelephoneNumber
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PartnerTelephoneNumberController @Inject()(val sessionService: SessionService,
+                                                 val authConnector: AuthConnector,
+                                                 val entityService: EntityService,
+                                                 val applicantDetailsService: ApplicantDetailsService,
+                                                 val vatRegistrationService: VatRegistrationService,
+                                                 view: PartnerTelephoneNumber)
+                                                (implicit appConfig: FrontendAppConfig,
+                                                 val executionContext: ExecutionContext,
+                                                 baseControllerComponents: BaseControllerComponents)
+  extends BaseController with SessionProfile with PartnerIndexValidation {
+
+  def show(index: Int): Action[AnyContent] = isAuthenticatedWithProfile() {
+    implicit request =>
+      implicit profile =>
+        validateIndex(index, routes.PartnerTelephoneNumberController.show) { optEntity =>
+          val form = optEntity.flatMap(_.telephoneNumber)
+            .fold(PartnerTelephoneForm.form)(PartnerTelephoneForm.form.fill)
+          optEntity match {
+            case Some(entity) if entity.displayName.isDefined =>
+              Future.successful(Ok(view(form, index, entity.displayName)))
+            case _ =>
+              logger.warn("[PartnerTelephoneNumberController] Attempted to capture telephone number without partyType")
+              Future.successful(Redirect(routes.PartnerEntityTypeController.showPartnerType(index)))
+          }
+        }
+  }
+
+  def submit(index: Int): Action[AnyContent] = isAuthenticatedWithProfile() {
+    implicit request =>
+      implicit profile =>
+        validateIndexSubmit(index, routes.PartnerTelephoneNumberController.show) {
+          PartnerTelephoneForm.form.bindFromRequest().fold(
+            formWithErrors =>
+              vatRegistrationService.partyType.flatMap {
+                case Individual => applicantDetailsService.getTransactorApplicantName.map {
```

Comment:
We're interested in the entity's partyType and displayName, not the applicant's

Reply:
Ah, I missed this bit, need to call `Entity.displayName` same like used in `show()`